### PR TITLE
SISRP-18468 Expand My Academics class schedules to include one-time meetings

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.berkeley.edu/browse/SISRP-18468

Back end at the top, front end at the bottom, and a regrettably large update of tests and test data (including getting all the different My Academics dummies back into alignment) in the middle.

### CODE REVIEW

**Comment by redconfetti** [comment]
LGTM

File: public/dummy/json/academics.json [hunk 1]
```diff
@@ -592,13 +604,16 @@
...
-                  "buildingName": "PIMENTAL",
-                  "roomNumber": "1",
-                  "schedule": "M 2:00P-3:00P"
-                }
-              ]
+              "schedules": {
+                "oneTime": [],
+                "recurring": [
+                  {
+                    "buildingName": "PIMENTAL",
```
Comment (johncrossman) [inline public/dummy/json/academics.json]: Should be 'Pimentel'. Not @paulkerschen's fault but perhaps he'll fix. :)

File: public/dummy/json/academics.json [hunk 2]
```diff
@@ -570,13 +579,16 @@
...
-                  "buildingName": "VALLEY LSB",
-                  "roomNumber": "2030",
-                  "schedule": "M 4:00P-5:00P"
-                }
-              ],
+              "schedules": {
+                "oneTime": [],
+                "recurring": [
+                  {
+                    "buildingName": "VALLEY LSB",
```
Comment (johncrossman) [inline public/dummy/json/academics.json]: @paulkerschen If you're making amendments then maybe we change this to 'Valley Life Sciences' as seen in EDO db.

File: public/dummy/json/academics_transition_future_profile.json [hunk 3]
```diff
@@ -469,11 +469,15 @@
...
       "timeBucket": "future",
+      "hasEnrollmentData": true,
       "campusSolutionsTerm": false,
       "classes": [
         {
+          "courseCatalog": "1A",
           "course_code": "BIOLOGY 1A",
+          "course_id": "biology-1a-2014-B",
           "dept": "BIOLOGY",
+          "dept_desc": "Biology",
```
Comment (johncrossman) [inline public/dummy/json/academics_transition_future_profile.json]: @paulkerschen mix of snake_case and camelCase is expected?

**Comment by johncrossman** [comment]
Looks good, @paulkerschen  I'll wait for your reply on minor typo and then merge when green Travis.

File: public/dummy/json/academics_transition_future_profile.json (hunk 3 above)
Comment (paulkerschen) [inline public/dummy/json/academics_transition_future_profile.json]: Yes, @johncrossman, despite the best efforts of our front-end devs, that is the current state of the feed.

**Comment by paulkerschen** [comment]
Thanks @johncrossman - amended to bring building names into our century.

**Pull request merged**